Source/Charts/Charts/PieChartView.swift

```diff
@@ -138,25 +158,88 @@ open class PieChartView: PieRadarChartViewBase
         {
             return
         }
+
+        let radius = adjustedRadius
         
-        let radius = diameter / 2.0
-        
-        let c = self.centerOffsets
+        let c = adjustedCenterOffsets()
         
         let shift = (data as? PieChartData)?.dataSet?.selectionShift ?? 0.0
         
         // create the circle box that will contain the pie-chart (the bounds of the pie-chart)
         _circleBox.origin.x = (c.x - radius) + shift
         _circleBox.origin.y = (c.y - radius) + shift
-        _circleBox.size.width = diameter - shift * 2.0
-        _circleBox.size.height = diameter - shift * 2.0
+        _circleBox.size.width = radius * 2 - shift * 2.0
+        _circleBox.size.height = radius * 2 - shift * 2.0
+
     }
-    
+
     internal override func calcMinMax()
     {
         calcAngles()
     }
-    
+
+    @objc open override func angleForPoint(x: CGFloat, y: CGFloat) -> CGFloat
+    {
+        let c = adjustedCenterOffsets()
+
+        let tx = Double(x - c.x)
+        let ty = Double(y - c.y)
+        let length = sqrt(tx * tx + ty * ty)
+        let r = acos(ty / length)
+
+        var angle = r.RAD2DEG
```

Comment:
It is probably worth making an `Angle` type in the future much like [SwiftUI.Angle](https://developer.apple.com/documentation/swiftui/angle)

Reply:
yep. but currently this should serve